src/server/repositories/UrlRepository.ts

```diff
@@ -604,6 +632,36 @@ export class UrlRepository implements UrlRepositoryInterface {
       }
     })
   }
+
+  public async updateSafeBrowsingExpiry(
+    shortUrl: string,
+    expiry: Date,
+  ): Promise<void> {
+    const url = await Url.findOne({ where: { shortUrl } })
+    if (!url) {
+      throw new NotFoundError(`Url with shortUrl ${shortUrl} not found`)
+    }
+
+    await this.update(
+      { shortUrl },
+      { safeBrowsingExpiry: expiry.toISOString() },
+      undefined,
+    )
```

Comment:
[question] the `update` method invalidates our redis cache, so now new accesses to get the expiry time would also hit our db, ensuring the latest `expiry` right?

Reply:
Yup that's correct, it purges the record from the redis cache so the next retrieval will need to go to DB, then get cached in redis with the new expiry time.